### ISSUE TITLE
[BUG]API changes in 0.6 causing KNN_Demo errors

### ISSUE THREAD

**Issue by taureandyernv** [rapidsai/notebooks#68]
**Describe the bug**
KNN demo has some challenges with RAPIDS Version 0.6

Bug 1:  In the second to last cell, 
```%%time
knn_cuml = cumlKNN(n_gpus=1)
knn_cuml.fit(X)
D_cuml,I_cuml = knn_cuml.query(X,n_neighbors)
```
Knn_demo `cumlKNN(n_gpus=1)` throws error

``` TypeError                                 Traceback (most recent call last)
<timed exec> in <module>

cuML/knn/knn_wrapper.pyx in cuml.KNN.__cinit__()

TypeError: __cinit__() got an unexpected keyword argument 'n_gpus'
```
I tested it with `cumlKNN(1)` and it seemed to work.  will do a PR

Bug 2: In the last cell, array_equal check fails despite being equal.
```passed = array_equal(D_sk,D_cuml)
message = 'compare knn: cuml vs sklearn distances %s'%('equal'if passed else 'NOT equal')
print(message)
passed = array_equal(I_sk,I_cuml)
message = 'compare knn: cuml vs sklearn indexes %s'%('equal'if passed else 'NOT equal')
print(message)
print("SKL shape: ", I_sk.shape, "; \tcuML shape: ",(len(I_cuml), len(I_cuml.columns))) ### added by me to confirm dimensions
```

```Output:
compare knn: cuml vs sklearn distances equal
compare knn: cuml vs sklearn indexes NOT equal
SKL shape:  (65536, 10) ; 	cuML shape:  (65536, 10)
```

**Comment by cjnolet** [comment]
Taurean,

Thank you for spotting this! We need to be better at updating the notebooks as we make changes to the cuml code. 

The n_gpus property has actually been removed permanently.

The multi-gpu support for kNN is in the process of being moved to Dask cuml. There is a single (optional) property on the constructor of the kNN object to downcast any doubles to floats, since the kNN backend currently does not have native support for double precision. 

I believe by using ‘1’ as the argument to the constructor, you are turning on the down-casting option here. It is not a bad idea to do this, but it might be more clear to users to use the full syntax: ‘should_downcast = True’

**Comment by cjnolet** [comment]
I believe I have covered many of these issues in #72. The one problem that remains is that the indices matrix appears to get further divergent from sklearn as data grows large.

I believe what’s happening is that items are close enough together that the results become non-deterministic. The distances match very closely, so I believe it’s not a reason for concern.

**Comment by taureandyernv** [comment]
closed as this was fixed in 0.6
